public/app/plugins/panel/bargauge/BarGaugePanel.tsx

```diff
@@ -1,27 +1,24 @@
-// Libraries
 import React, { PureComponent } from 'react';
-
-// Services & Utils
-import { config } from 'app/core/config';
-
-import { BarGauge, VizRepeater, VizRepeaterRenderValueProps, DataLinksContextMenu } from '@grafana/ui';
-import { BarGaugeOptions } from './types';
 import {
-  getFieldDisplayValues,
+  DisplayValueAlignmentFactors,
   FieldDisplay,
-  PanelProps,
   getDisplayValueAlignmentFactors,
-  DisplayValueAlignmentFactors,
+  getFieldDisplayValues,
+  PanelProps,
 } from '@grafana/data';
+import { BarGauge, DataLinksContextMenu, VizRepeater, VizRepeaterRenderValueProps } from '@grafana/ui';
+
+import { config } from 'app/core/config';
+import { BarGaugeOptions } from './types';
 
 export class BarGaugePanel extends PureComponent<PanelProps<BarGaugeOptions>> {
   renderValue = (valueProps: VizRepeaterRenderValueProps<FieldDisplay, DisplayValueAlignmentFactors>): JSX.Element => {
     const { options } = this.props;
     const { value, alignmentFactors, orientation, width, height } = valueProps;
-    const { field, display, view, colIndex } = value;
+    const { field, display, view, colIndex, hasLinks, getLinks } = value;
 
     return (
-      <DataLinksContextMenu links={value.getLinks}>
+      <DataLinksContextMenu links={getLinks} hasLinks={hasLinks}>
```

Comment:
Hm, I'm not a fan of boolean props to decide whether or not something should be rendered. I would go one step further here and probably make DataLinksContextMenu `links` property non optional and only wrap in Context menu if the links are there. WDYT?

Reply:
Yes, I started down this path but then that adds another if statement for each component and they grow but again I like the quick exit strategy. So I agree and I will close https://github.com/grafana/grafana/pull/24232